File: HashMap/001.two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i< nums.length; i++){
            int j = target- nums[i]; //==7 
            if (map.containsKey(j)){ 
                int[] intArray = new int[] {i, map.get(j)};
                return intArray; 

            }else {
                map.put(nums[i], i); 
            }
        }
        int[] notfound = new int[] {};
        return notfound;
    }
}

File: Array/001.two-sum.js
const twoSum = (arr, target) => {
    const VAL_MAP = {};
    for (let i = 0; i < arr.length; i++) {
        let diff = target - arr[i];
        if (VAL_MAP[diff] !== undefined) {
            return [VAL_MAP[diff], i];
        } else {
            VAL_MAP[arr[i]] = i;
        }
    }
}

const nums = [2, 7, 11, 15];
console.log(twoSum(nums, 9));